lib/segment/src/index/field_index/full_text_index/text_index.rs

```diff
@@ -197,22 +197,48 @@ impl FullTextIndex {
         }
     }
 
-    pub fn check_match(
+    /// Build the set intersection of point ids from the query tokens
+    pub fn query_tokens_posting_intersection(
```

Comment:
That's a mouthful of terms 😅 
```suggestion
    pub fn collect_intersection(
```